test/uk/gov/hmrc/pillar2/service/SubscriptionServiceSpec.scala

```diff
@@ -89,6 +90,50 @@ class SubscriptionServiceSpec extends BaseSpec with Generators with ScalaCheckPr
       }
 
     }
+
+    "handle LimitedLiabilityPartnership entity type correctly" - {
+      "when all required data is present" in {
+        val userAnswersGen = userAnswersFromGenerators(arbitraryWithIdUpeFmUserDataLLP)
+
+        when(
+          mockSubscriptionConnector
+            .sendCreateSubscriptionInformation(any[RequestDetail]())(any[HeaderCarrier](), any[ExecutionContext]())
+        ).thenReturn(
+          Future.successful(
+            HttpResponse.apply(OK, "Success")
+          )
+        )
+
+        forAll(arbitrary[String], Gen.option(arbitrary[String]), userAnswersGen.arbitrary) { (upeSafeId, fmSafeId, userAnswers) =>
+          service.sendCreateSubscription(upeSafeId, fmSafeId, userAnswers).map { response =>
+            response.status mustBe OK
+          }
+        }
+      }
+
+      "throw exception when company profile is missing" in {
+        val llpWithoutCompanyProfile = arbitraryWithIdUpeFmUserDataLLP.arbitrary
+        val userAnswersGen           = userAnswersFromGenerators(Arbitrary(llpWithoutCompanyProfile))
+
+        forAll(arbitrary[String], Gen.option(arbitrary[String]), userAnswersGen.arbitrary) { (upeSafeId, fmSafeId, userAnswers) =>
+          val modifiedData = (userAnswers.data \ "upeGRSResponse" \ "partnershipEntityRegistrationData").as[JsObject] - "companyProfile"
```

Comment:
Doing this rather than doing the quite verbose task of creating a whole new generator just for removing company profile